test/integration/virtualnode/virtualnode_test.go

```diff
@@ -146,5 +147,93 @@ var _ = Describe("VirtualNode", func() {
 			})
 
 		})
+
+		It("should delete a virtual node in AWS", func() {
+
+			meshName := fmt.Sprintf("%s-%s", f.Options.ClusterName, utils.RandomDNS1123Label(6))
+			mesh := &appmesh.Mesh{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: meshName,
+				},
+				Spec: appmesh.MeshSpec{
+					NamespaceSelector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"mesh": meshName,
+						},
+					},
+				},
+			}
+
+			By("creating a mesh resource in k8s", func() {
+				err := meshTest.Create(ctx, f, mesh)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			By("validating the resources in AWS", func() {
+				err := meshTest.CheckInAWS(ctx, f, mesh)
+				Expect(err).NotTo(HaveOccurred())
+
+			})
+
+			By("Create a namespace and add labels", func() {
+				namespace, err := f.NSManager.AllocateNamespace(ctx, "appmeshtest")
+				Expect(err).NotTo(HaveOccurred())
+				vnBuilder.Namespace = namespace.Name
+				vnTest.Namespace = namespace
+
+				oldNS := namespace.DeepCopy()
+				namespace.Labels = algorithm.MergeStringMap(map[string]string{
+					"appmesh.k8s.aws/sidecarInjectorWebhook": "enabled",
+					"mesh":                                   meshName,
+				}, namespace.Labels)
+
+				err = f.K8sClient.Patch(ctx, namespace, client.MergeFrom(oldNS))
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			vnName := fmt.Sprintf("vn-%s", utils.RandomDNS1123Label(8))
+			listeners := []appmesh.Listener{vnBuilder.BuildListener("http", 8080)}
+			backends := []types.NamespacedName{}
+			vn := vnBuilder.BuildVirtualNode(vnName, backends, listeners)
+
+			By("Creating a virtual node resource in k8s", func() {
+				err := vnTest.Create(ctx, f, vn)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			By("Validating the virtual node in AWS", func() {
+				err := vnTest.CheckInAWS(ctx, f, mesh, vn)
+				Expect(err).NotTo(HaveOccurred())
+
+			})
+
+			By("Check mesh finalizers", func() {
+				var wg sync.WaitGroup
+				wg.Add(1)
+
+				go func() {
+					meshTest.Cleanup(ctx, f)
+					wg.Done()
+				}()
+
```

Comment:
wg.Wait() is added later on line # 247 since we want to let this run in the background, check there's deletion timestamp, verify there's finalizers (i.e. mesh has them), verify the dependencies are there (vn is there), delete vn, verify again it's gone,  do wg.Wait()and verify mesh is actually deleted. Deletion of virtual node would have allowed the deletion of mesh to move forward